pkg/hub/client/public/type_rev_fields.go

```diff
@@ -0,0 +1,33 @@
+package public
+
+import "fmt"
+
+// typeRevisionFieldsRegistry holds possible fields configuration for InterfaceRevision query.
+var typeRevisionFieldsRegistry = map[TypeRevisionQueryFields]string{
+	TypeRevisionRootFields: `
+		revision`,
+	TypeRevisionMetadataFields:          typeRevisionMetadataFields,
+	TypeRevisionSpecFields:              typeRevisionSpecFields,
+	TypeRevisionSpecAdditionalRefsField: typeRevisionSpecAdditionalRefsField,
+}
+
+// typeRevisionMetadataFields for querying TypeRevision's Metadata fields.
+var typeRevisionMetadataFields = fmt.Sprintf(`
+      metadata {
+        %s
+      }`, genericMetadataFields)
+
+// typeRevisionSpecFields for fetching TypeRevision's spec fields only.
+var typeRevisionSpecFields = `
+      spec {
+        jsonSchema
+        additionalRefs
+      }
+`
+
+// typeRevisionSpecAdditionalRefsField for fetching TypeRevision's spec.additionalRefs field only.
+var typeRevisionSpecAdditionalRefsField = `
+      spec {
+        additionalRefs
+      }
```

Comment:
nit: Please use const whenever is possible, and also follow the convention for writing comments:
```suggestion
// typeRevisionMetadataFields  specifies fields  for querying TypeRevision's Metadata fields.
var typeRevisionMetadataFields = fmt.Sprintf(`
      metadata {
        %s
      }`, genericMetadataFields)

// typeRevisionSpecFields specifies fields for fetching TypeRevision's spec fields only.
const typeRevisionSpecFields = `
      spec {
        jsonSchema
        additionalRefs
      }
`

// typeRevisionSpecAdditionalRefsField specifies fields for fetching TypeRevision's spec.additionalRefs field only.
const typeRevisionSpecAdditionalRefsField = `
      spec {
        additionalRefs
      }
```